R/wb_functions.R

```diff
@@ -393,75 +393,97 @@ wb_to_df <- function(
   keep_rows <- keep_rows[keep_rows %in% rnams]
 
   # reduce data to selected cases only
-  cc <- cc[cc$row_r %in% keep_rows & cc$c_r %in% keep_cols, ]
+  if (!is.null(cols) && !is.null(rows) && !missing(dims))
+    cc <- cc[cc$row_r %in% keep_rows & cc$c_r %in% keep_cols, ]
 
   # if (!nrow(cc)) browser()
 
   cc$val <- NA
   cc$typ <- NA
 
+  cc_tab <- table(cc$c_t)
+
   # bool
-  sel <- cc$c_t %in% c("b")
-  cc$val[sel] <- as.logical(as.numeric(cc$v[sel]))
-  cc$typ[sel] <- "b"
+  if (isTRUE(cc_tab[c("b")] > 0)) {
```

Comment:
in the table there might not be a column "b", therefore enforce this check with isTRUE